src/SlackClientFactory.ts

```diff
@@ -52,10 +62,33 @@ export class SlackClientFactory {
      */
     public async getTeamClient(teamId: string): Promise<WebClient> {
         if (this.teamClients.has(teamId)) {
-            return this.teamClients.get(teamId)!;
+            const set = this.teamClients.get(teamId);
+            // Check the auth on the client every AUTH_INTERVAL_MS, and if it fails, refetch the team.
+            if (set && Date.now() - set.lastTestTs < AUTH_INTERVAL_MS) {
+                // set has not expired
+                return set.client;
+            } else if (set) {
```

Comment:
Is there a condition where `this.teamClients.get(teamId)` could return null? Why are we checking `set` here?

Reply:
Yep, if the team hasn't been created yet, it will be null.